### PR TITLE
Fix NullReferenceException thrown in PopupRenderer.SetViewController

### LINKED ISSUE #1758: NullReferenceException thrown in PopupRenderer.SetViewController when using in embedded forms scenario
### Description

I am trying to use the Popup View in a Xamarin.iOS application embedding some Xamarin.Forms views, a technique described in the Xamarin.Forms documentation here: https://docs.microsoft.com/en-us/xamarin/xamarin-forms/platform/native-forms

When calling `Xamarin.Forms.Application.Current.MainPage.Navigation.ShowPopup(popup);` I get a null reference exception, as shown in the stack trace.

### Stack Trace

```shell
System.NullReferenceException: Object reference not set to an instance of an object
  at Xamarin.CommunityToolkit.UI.Views.PopupRenderer.SetViewController () [0x00089] in <be1d33109c044cbba821f6c39d395abb>:0 
  at Xamarin.CommunityToolkit.UI.Views.PopupRenderer.OnElementChanged (Xamarin.Forms.Platform.iOS.ElementChangedEventArgs`1[TElement] e) [0x0001f] in <be1d33109c044cbba821f6c39d395abb>:0 
  at Xamarin.CommunityToolkit.UI.Views.PopupRenderer.Xamarin.Forms.Platform.iOS.IVisualElementRenderer.SetElement (Xamarin.Forms.VisualElement element) [0x0008d] in <be1d33109c044cbba821f6c39d395abb>:0 
  at Xamarin.Forms.Platform.iOS.Platform.CreateRenderer (Xamarin.Forms.VisualElement element) [0x00032] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Platform.cs:240 
  at Xamarin.CommunityToolkit.Extensions.NavigationExtensions.PlatformShowPopup (Xamarin.CommunityToolkit.UI.Views.BasePopup popup) [0x00000] in <be1d33109c044cbba821f6c39d395abb>:0 
  at Xamarin.CommunityToolkit.Extensions.NavigationExtensions.ShowPopup (Xamarin.Forms.INavigation navigation, Xamarin.CommunityToolkit.UI.Views.BasePopup popup) [0x00000] in <be1d33109c044cbba821f6c39d395abb>:0 
  at XctEmbedDemo.MainViewController.OnTouchUpInside (System.Object sender, System.EventArgs e) [0x0000d] in /Users/follesoe/Projects/XctEmbedDemo/XctEmbedDemo/MainViewController.cs:51
```

### Link to Reproduction Sample

I have created a minimal Xamarin.iOS application referencing the latest builds of Xamarin.Forms and Xamarin.CommunityToolkit that will trigger this issue: https://github.com/follesoe/XctEmbedDemo

### Steps to Reproduce

1. Run the example application
2. Tap the button
3. Observe the null reference exception

### Expected Behavior

I would expect the popup to appear, or at least get a meaningful error saying I am using the component in an unsupported way.

### Actual Behavior

Null reference exception is thrown.

### Basic Information

- Version with issue: 1.3.1
- Last known good version:
- IDE:
- Platform Target Frameworks: 
  - iOS:  15.0, 
  - Xamarin.iOS: 15.2.0.17 
  - Android:  
  - UWP:   
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

### PR DESCRIPTION
**Pull request by VladislavAntonyuk**
### Description of Bug ###

1. Fix NRE when renderer if currentPageRenderer is null
2. Allow using Popup in Xamarin Native

### Issues Fixed ###

- Fixes #1758 

### PR Checklist ###

- [ ] Has a linked Issue, and the Issue has been `approved`
- [ ] Has tests (if omitted, state reason in description)
- [ ] Has samples (if omitted, state reason in description)
- [ ] Rebased on top of main at time of PR
- [ ] Changes adhere to coding standard

- [ ] Updated [documentation](https://github.com/MicrosoftDocs/xamarin-communitytoolkit)

### CODE REVIEW

**Review by ahoefling — APPROVED** [review]
This looks to me, thanks for making the change so quickly. 🚀🚀

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/iOS/PopupRenderer.ios.cs [hunk 1]
```diff
@@ -135,17 +141,26 @@ void SetViewController()
...
+			}
+
+			if (currentPageRenderer == null)
+			{
+				ViewController ??= page.CreateViewController();
 			}
 			else
 			{
-				currentPageRenderer = Platform.GetRenderer(mainPage);
+				ViewController ??= currentPageRenderer.ViewController;
```
Comment (pictos) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/iOS/PopupRenderer.ios.cs:162]: Are we sure that we just want to set the ViewController if it's null?

**Review by pictos — COMMENTED** [review]
I've a question

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/iOS/PopupRenderer.ios.cs (hunk 1 above)
Comment (VladislavAntonyuk) [reply src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/iOS/PopupRenderer.ios.cs:162]: previously, we always set it. it was done by CreateControl. it was the only 1 place where we set it.
now we also set ViewController in ctor. we do not want to set it here again.
hope I correctly understand your question.

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/iOS/PopupRenderer.ios.cs [hunk 2]
```diff
@@ -135,17 +141,26 @@ void SetViewController()
 		{
 			IVisualElementRenderer currentPageRenderer;
 			var modalStackCount = Application.Current.MainPage?.Navigation?.ModalStack?.Count ?? 0;
-			var mainPage = Application.Current.MainPage;
+			var page = Application.Current.MainPage;
 			if (modalStackCount > 0)
 			{
 				var index = modalStackCount - 1;
-				currentPageRenderer = Platform.GetRenderer(mainPage!.Navigation!.ModalStack![index]);
+				page = page!.Navigation!.ModalStack![index];
```
Comment (brminnick) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/iOS/PopupRenderer.ios.cs:148]: Let's avoid using the Null-Forgiving Operator, `!`. 

It's a code smell that'll just lead to future bugs.

**Review by brminnick — CHANGES_REQUESTED** [review]
Thanks Vlad! 

Let's remove the [Null Forgiving Operator](https://docs.microsoft.com/dotnet/csharp/language-reference/operators/null-forgiving)

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/iOS/PopupRenderer.ios.cs [hunk 3]
```diff
@@ -140,12 +140,12 @@ void CreateControl()
...
 		{
 			IVisualElementRenderer currentPageRenderer;
-			var modalStackCount = Application.Current.MainPage?.Navigation?.ModalStack?.Count ?? 0;
 			var page = Application.Current.MainPage;
+			var modalStackCount = page?.Navigation?.ModalStack?.Count ?? 0;
 			if (modalStackCount > 0)
 			{
 				var index = modalStackCount - 1;
-				page = page!.Navigation!.ModalStack![index];
+				page = page?.Navigation.ModalStack[index];
```
Comment (pictos) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/iOS/PopupRenderer.ios.cs]: if `ModalStack` is > 0 `Page` will not be null

Comment (VladislavAntonyuk) [reply src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/iOS/PopupRenderer.ios.cs]: I know, but we receive an error if remove null propagation. and we discussed not to use Null forgiving operator
image

Comment (VladislavAntonyuk) [reply src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/iOS/PopupRenderer.ios.cs]: rewrite it and add check page for null

**Review by pictos — APPROVED** [review]

**Pull request merged**